clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -522,6 +522,80 @@ static Expected<StringRef> convertSPIRVToIR(StringRef Filename,
   return *TempFileOrErr;
 }
 
+// Update sycl-post-link options based on target triple.
+static void updateCmdArgs(SmallVector<StringRef, 8> &CmdArgs,
+                          llvm::Triple Triple) {
+  // Get an argument in CmdArgs that contains Str. If there is no such
+  // argument, an empty argument is returned
+  auto getArg = [&](const StringRef &Str) {
+    for (auto Arg : CmdArgs)
+      if (Arg.contains(Str))
+        return Arg;
+    return StringRef("");
+  };
+  // Add a new argument Arg to CmdArgs if not present already.
+  auto addArg = [&](const StringRef &Arg) {
+    auto ArgPresent = (getArg(Arg) != "");
+    if (!ArgPresent)
+      CmdArgs.push_back(Arg);
+  };
+  // Replace an argument in CmdArgs that contains Str with NewArg. If no such
+  // argument is present, add the NewArg to CmdArgs.
+  auto replaceOrAddArg = [&](const StringRef &NewArg, const StringRef &Str) {
+    for (auto &Arg : CmdArgs)
+      if (Arg.contains(Str)) {
+        Arg = NewArg;
+        return;
+      }
+    CmdArgs.push_back(NewArg);
+  };
+  // Remove argument containing Str from CmdArgs.
+  auto removeArg = [&](const StringRef &Str) {
+    CmdArgs.erase(
+        std::remove_if(CmdArgs.begin(), CmdArgs.end(),
+                       [&](StringRef Arg) { return Arg.contains(Str); }),
+        CmdArgs.end());
+  };
+
+  // specialization constants processing.
+  bool IsAOT = Triple.isNVPTX() || Triple.isAMDGCN() || Triple.isSPIRAOT();
+  if (!IsAOT)
+    replaceOrAddArg("-spec-const=native", "-spec-const");
+  else
+    replaceOrAddArg("-spec-const=emulation", "-spec-const");
+
+  // -emit-only-kernels-as-entry-points is set by user and is enabled only for
+  // Intel targets.
+  auto EmitOnlyKernelsAsEntryPointsArg =
+      getArg("-emit-only-kernels-as-entry-points");
+  if ((EmitOnlyKernelsAsEntryPointsArg != "") && !Triple.isNVPTX() &&
+      !Triple.isAMDGPU())
+    addArg("-emit-only-kernels-as-entry-points");
+  else
+    removeArg("-emit-only-kernels-as-entry-points");
+
+  if (!(Triple.isAMDGCN()))
+    addArg("-emit-param-info");
+
+  if (Triple.isNVPTX() || Triple.isAMDGCN())
+    addArg("-emit-program-metadata");
+
+  if (Triple.isSPIROrSPIRV()) {
+    addArg("-symbols");
+    addArg("-emit-exported-symbols");
+    addArg("-split-esimd");
+    addArg("-lower-esimd");
+  }
+
+  // Here, IsAOT includes x86_64 device as well.
+  IsAOT = IsAOT || Triple.getSubArch() == llvm::Triple::SPIRSubArch_x86_64;
```

Comment:
I would make a new variable instead of changing what is represents, maybe the first variable is
`IsAOTNotFPGA` and this one is `IsAOT` or something